### PR TITLE
Land the test cases that should've landed with the selectors update

### LINKED ISSUE #15448: Land the test cases that should've landed with the selectors update
See https://github.com/servo/servo/issues/15335 and https://github.com/servo/servo/issues/15394.

### PR DESCRIPTION
**Pull request by mhaessig**
PR for #15448
- Added test for pseudo-element with whitespace #15335
- Added tests for empty values in `[att^=""]`, `[att$=""]` and `[att*=""]` selectors #15394 (tests for `[att~=""]` have been added with #15440)

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15448
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18542)

### CODE REVIEW

File: tests/wpt/web-platform-tests/dom/nodes/selectors.js [hunk 1]
```diff
@@ -119,6 +120,7 @@ var validSelectors = [
   // - substring begins-with     [att^=val] (Level 3)
   {name: "Attribute begins with selector, matching href attributes beginning with specified substring",                             selector: "#attr-begins a[href^=\"http://www\"]", expect: ["attr-begins-a1", "attr-begins-a3"],     level: 3, testType: TEST_QSA | TEST_MATCH},
   {name: "Attribute begins with selector, matching lang attributes beginning with specified substring, ",                           selector: "#attr-begins [lang^=\"en-\"]",         expect: ["attr-begins-div2", "attr-begins-div4"], level: 3, testType: TEST_QSA | TEST_MATCH},
+  {name: "Attribute begins with selector, not matching class attribute with empty value",                                           selector: "#attr-begins [class^= \"\"]",          expect: [] /*no matches*/,                        level: 3, testType: TEST_QSA},
```
Comment (emilio) [inline tests/wpt/web-platform-tests/dom/nodes/selectors.js]: nit: I'd remove the space in `= \"` for consistency.

File: tests/wpt/web-platform-tests/dom/nodes/selectors.js [hunk 2]
```diff
@@ -488,6 +492,7 @@ var scopedSelectors = [
   // - substring begins-with     [att^=val] (Level 3)
   {name: "Attribute begins with selector, matching href attributes beginning with specified substring",                             selector: "a[href^=\"http://www\"]", ctx: "#attr-begins", expect: ["attr-begins-a1", "attr-begins-a3"],     level: 3, testType: TEST_FIND | TEST_MATCH},
   {name: "Attribute begins with selector, matching lang attributes beginning with specified substring, ",                           selector: "[lang^=\"en-\"]",         ctx: "#attr-begins", expect: ["attr-begins-div2", "attr-begins-div4"], level: 3, testType: TEST_FIND | TEST_MATCH},
+  {name: "Attribute begins with selector, not matching class attribute with empty value",                                           selector: "[class^= \"\"]",          ctx: "#attr-begins", expect: [] /*no matches*/,                        level: 3, testType: TEST_FIND},
```
Comment (emilio) [inline tests/wpt/web-platform-tests/dom/nodes/selectors.js]: ditto here.

**Review by emilio — APPROVED** [review]
Thanks so much! This looks good to me, with those two nits fixed.

**Comment by mhaessig** [comment]
Removed the spaces.

**Comment by emilio** [comment]
Could you squash your commits? Also, your last commit means that the hash in the manifest should change, you can run `./mach update-manifest` for that.

**Comment by emilio** [comment]
(I'm fine with two separate commits if you want, but the `fixup!` one should be squashed with the commit introducing those tests)

**Comment by mhaessig** [comment]
Will do

**Comment by emilio** [comment]
Thank you so much! :)

**Comment by mhaessig** [comment]
I hope I didn't screw up the hash in the manifest by squashing after running `./mach update-manifest`.

**Comment by emilio** [comment]
Should be fine, thanks again!

@bors-servo r+

**Pull request merged**